### PR TITLE
feat: auto-sync npm plugins from typeclaw.json into package.json

### PR DESCRIPTION
**Pull request by devxoul**
## Background

Today a user must declare an npm plugin in **both** `typeclaw.json#plugins` *and* `package.json#dependencies` — the loader only ever reads `node_modules`, so the two files must be kept in sync by hand. There is no enforcement; they drift silently and the mismatch is only discovered at agent boot.

## Summary

Makes `typeclaw.json#plugins` the **single source of truth** for npm-distributed plugins. After this PR, the user edits only `typeclaw.json`; `typeclaw start` materializes npm entries into `package.json` and installs them.

```json
{
  "plugins": ["typeclaw-plugin-standup@1.2.3", "@acme/typeclaw-plugin-helicone"]
}
```

Removing an entry prunes it. A bare name (no `@version`) resolves to the current latest and is pinned, exactly like `bun add <name>`. The resting state stays the clean Bun-native `package.json` + `bun.lock` layout.

### 3 atomic commits (dependency order)

1. **`feat(plugin): parse name[@version]`** — `splitPluginEntrySpec` splits an entry into name + optional version (scoped packages split on the last `@`). The loader resolves/imports under the bare name, so `typeclaw-plugin-foo@1.2.3` still finds `node_modules/typeclaw-plugin-foo`.
2. **`feat(init): materialize plugins into package.json on start`** — `reconcilePluginDeps` runs before `ensureDeps` so the existing drift detector installs the new deps in the same pass. Bidirectional: adds/updates what the config lists, prunes what it previously managed. Provenance under `package.json#typeclaw.managedPlugins` ensures the prune **only ever touches deps this step added** — never user or runtime deps. Reconcile failures abort `start` cleanly.
3. **`feat(security): gate plugin additions behind owner/trusted`** — new `pluginAddition` guard (medium tier; owner+trusted bypass).

### Security model

Adding/version-bumping a plugin in `typeclaw.json` is a **deferred host-code-execution grant**: the entry is installed by the next host `typeclaw start`, where npm lifecycle scripts run as the operator. The agent can write `/agent/typeclaw.json` under prompt injection, so the dangerous write is gated **where role context exists** (the container `tool.before` boundary), by the same mechanism as the existing `rolePromotion` / `cronPromotion` guards:

- **Blocked for member/guest** (and prompt-injected low-trust input): adding a new plugin entry, or changing an existing entry's version. Ack-overridable for genuine operator intent.
- **Allowed for everyone**: removal, reorder, and local-path entries (`./`, `../`, absolute) — none of these install registry code.

Because the write is trust-gated upstream, the host-side reconcile runs **seamlessly with no prompts** — the `plugins` field is trustworthy by construction. The role model can't gate the host install directly because that happens later, outside channel/session provenance — so we gate the write instead.

## Preview

None — no UI or output changes.

## What this does NOT do

- Does not affect local-path plugin entries (`./`, `../`, absolute paths) — those are never touched by the reconciler.
- Does not change the loader's resolution contract: plugins are always imported under the bare package name regardless of whether a version suffix was declared.
- Does not add any interactive prompts on the host side; the security gate lives exclusively in the container at write time.

## Review notes

- Load-bearing change: `reconcilePluginDeps` in `src/init/reconcile-plugin-deps.ts` and its call site in `src/container/start.ts` (runs before `ensureDeps`). The provenance key under `typeclaw.managedPlugins` in `package.json` is what makes pruning safe — it scopes removals to only what this step ever wrote.
- Security invariant to verify: `src/bundled-plugins/security/policies/plugin-addition.ts` uses the same path-identity + diff + ack shape as `rolePromotion`; multi-edit is refused conservatively (no ack path) to prevent array-rewrite tricks.
- Tests: `src/plugin/loader.test.ts` (parse edge cases), `src/init/reconcile-plugin-deps.test.ts` (add/remove/update/idempotence/never-prunes-user-deps), `src/bundled-plugins/security/policies/plugin-addition.test.ts` (block/allow/ack matrix), `permissions.test.ts` drift guard updated for the new medium-tier entry. All checks green: `bun run typecheck`, `bun run lint` (0 errors), `bun run format:check`, `bun run test` (8455 pass / 0 fail).

### CODE REVIEW

File: src/init/reconcile-plugin-deps.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+// pins it, mirroring `npm/bun add <name>` writing a concrete version.
+async function resolveDesiredManaged(
+  plugins: readonly string[],
+  resolveLatest: ResolveLatestVersion,
+): Promise<Record<string, string>> {
+  const desired: Record<string, string> = {}
+  for (const entry of plugins) {
+    if (isLocalEntry(entry)) continue
+    const { name, versionSpec } = splitPluginEntrySpec(entry)
+    if (name.length === 0) continue
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/init/reconcile-plugin-deps.ts:105]: Fixed in af37307d. A bare `plugins[]` entry now reuses the version already in `managedPlugins` and only resolves `latest` for a genuinely new, unmanaged plugin — so the pin is stable across starts and an unchanged `typeclaw.json` produces no `package.json` rewrite (no gate bypass). Added a test asserting `resolveLatest` is never called for an already-pinned bare entry.

File: src/container/start.ts [hunk 2]
```diff
@@ -247,6 +248,22 @@ export async function start({
...
+    // field at owner/trusted trust, so by the time this host-side step runs the
+    // field is trustworthy by construction.
+    const pluginReconcile = await reconcilePluginDeps({
+      cwd,
+      plugins: (await loadTypeclawConfig(cwd)).plugins,
+    }).catch((error: unknown) => ({ error: error instanceof Error ? error.message : String(error) }) as const)
+    if ('error' in pluginReconcile) {
+      return { ok: false, reason: `plugin dependency reconcile failed: ${pluginReconcile.error}` }
+    }
+
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/container/start.ts:267]: Fixed in 8340f733. `pluginReconcile.changed` is now ORed into the force-install decision, so any reconcile that rewrote `package.json` (managed version bump or removal) forces `bun install` — `bun.lock`/`node_modules` stay in sync even when the dependency dir already exists. Added two start tests: a managed version bump and a managed removal, both with the dep dir already present, asserting `ensureDeps` is called with `force: true`.
